pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -291,6 +307,15 @@ func (s *Service) handle(msg *message, aEvent chan<- service.DIDCommAction) erro
 		}
 	}
 
+	// notifies the introduce protocol that invitation was received
+	if msg.Msg.Header.Type == InvitationMsgType && invitation != nil {
+		s.inrcMutex.RLock()
+		if s.invitationReceivedCallback != nil {
+			go s.invitationReceivedCallback(invitation)
```

Comment:
This seems better modelled as a go channel, rather than spinning up a goroutine.